### DISCUSSION TITLE
How to use local modules?

### DISCUSSION THREAD

**Discussion by crotanite** [nuxt/nuxt#18243] [Questions] [answered]
There is documentation for local modules here

https://v3.nuxtjs.org/docs/directory-structure/modules

Does this mean creating a 'modules' directory?
If so, what name should each individual module have to be loaded in the nuxt config file?

**Comment by wobsoriano** [comment, marked as answer]
Nuxt doesn't autoload local modules created inside a modules directory unlike plugins but you can import it in nuxt config

```ts
// modules/your-module.ts
import { defineNuxtModule } from '@nuxt/kit'

export default defineNuxtModule({
  configKey: 'sample',
  setup (options) {
      console.log(options)
  },
})
```

```ts
import { defineNuxtConfig } from 'nuxt3';

export default defineNuxtConfig({
  modules: ['~/modules/your-module.ts'],
  sample: { foo: 'bar' }
});
```

**Reply by wobsoriano** [reply]
Hi @crotanite , did my answer help you? If yes, can you mark it as answered? Thanks

**Reply by crotanite** [reply]
Yes it did! Got caught up in building the module, that I forgot to accept this answer 😆 Thank you!
